### ISSUE TITLE
Question about descriptor loss

### ISSUE THREAD

**Issue by 904281665** [rpautrat/SuperPoint#164]
Hi, Sorry for disturbing you. I have some question about the descriptor loss, Hope to get your help!
 ```
   dot_product_desc = tf.reshape(tf.nn.l2_normalize(
        tf.reshape(dot_product_desc, [batch_size, Hc, Wc, Hc * Wc]),
        3), [batch_size, Hc, Wc, Hc, Wc])
    dot_product_desc = tf.reshape(tf.nn.l2_normalize(
        tf.reshape(dot_product_desc, [batch_size, Hc * Wc, Hc, Wc]),
        1), [batch_size, Hc, Wc, Hc, Wc])
```
In this position, We get dot_product_desc after l2 normal. But when we calculate the loss, we hope that the positive_dist is greater than 1. As we all know after l2 normal, the feature map cannot take the value 1. So in the training phase, the loss will not converge. Is my understanding wrong?I have been troubled by this problem for a long time.

**Comment by 904281665** [comment]
Thank you so much！ When the model is not pre-trained, adding l2 normal directly will cause the loss to be difficult to converge.
So I first use the descriptor loss in the paper to pre-train the model. Then I added l2 normal to the descriptor loss to fine-tune the model, and finally got the desired effect！

**Comment by rpautrat** [comment]
Thanks for sharing this insight!
